### DISCUSSION TITLE
Anyone figured out how to override templates in new hotspot manager of v8?

### DISCUSSION THREAD

**Discussion by mike240se** [Art-of-WiFi/UniFi-API-client#212] [Q&A]
Many of us are using this API client for custom captive portals.  I am using this API for an old custom system I created where everyone at my company is a guest and must be authorized before they can use the wifi.  (because of the need to provide quick and easy wpa2 password auth).   My system redirects to a web app i created that requires them to register with their name, email and department, etc and read a disclaimer and rules.  Once they do that it auto authorizes them.

However, I was doing that by setting Legacy JSP in guest control with override templates with custom changes.   Then in /usr/lib/unifi/data/sites/default/portal/index.html i had:

![image](https://github.com/Art-of-WiFi/UniFi-API-client/assets/31720459/1357151f-4762-462e-9cee-56e3e0ad81f8)

The problem is in verison 8 they redid the guest control to the new hotspot manager and it seems this is not possible.  

Has anyone found a workaround? I asked on the unifi forums but no ressponse yet.

PS. If anyone knows of a way to require a unifi client to provide some info (like name, email, etc) using the unifi network controller, I would be willing to just ditch my app and usse  that.

**Comment by malle-pietje** [comment]
You are not alone I'm afraid, especially those that use the custom `index.html` route for Radius-backed authentication through a captive portal are being affected by this because they have no alternative.

My reply on the community explained what you can do to change your code to work with the native redirect provided by the UniFi platform.

Alternatively you can always go for an off-the-shelf-solution like our captive portal 😉 
https://artofwifi.net/captive-portal-software-for-unifi

This highly flexible captive portal solution is available as software that we install on a server that you provide or as a **Captive Portal as a Service**. Feel free to ping me or email me (info AT artofwifi DOT net) for further details.

**Reply by mike240se** [reply]
According to Sloof on the unifi forums you can just use tthe one way portal redirect and it will send the same variables in the GET.   Why wouldnt that work for us if true?

**Reply by malle-pietje** [reply]
Slooffmaster is me😉 What isn’t clear from my answer on the community? Should come down to the exact same as I wrote above.

**Reply by mike240se** [reply]
Sorry I read this a while ago and then had left this open in my browser and came back and read the first paragraph and replied without reading the rest.  DOH, my bad.

So those that are using a custom index.html are not doing a simple redirect like I am doing I assume?   I was under the impression everyone was just using the index to fire off a redirect with some variables.

**Comment by malle-pietje** [comment]
No problem. The cases I described where the users were having issues, use the `index.html` for Radius auth (long story). For regular redirects the native method works fine as I explained.
